### PR TITLE
Specifying the value used for padding

### PR DESCRIPTION
**Pull request by hassyGo**
The "pad_packed_sequence" function fills padded elements with zeros, but sometimes it is not useful. For example, some previous papers on NLP, including my recent paper [1], use a max-pooling technique for RNN-based sentence representations. More specifically, the max-pooling technique selects the maximum value from all time steps (i.e., hidden states) for each dimension. In such a case, we do not want the padded zeros to be selected. To overcome this situation, we can simply use a very small value instead of zero.

An LSTM example is shown below:

input = embedding(Variable(batchInput))
packedInput = torch.nn.utils.rnn.pack_padded_sequence(input, lengths, batch_first = True)
h, (hn, cn) = lstm(packedInput, (h0, c0))
h, _ = torch.nn.utils.rnn.pad_packed_sequence(h, -1024.0, batch_first = True)
sentenceRep, _ = torch.max(h, 1, keepdim = True)

[1] A Joint Many-Task Model: Growing a Neural Network for Multiple NLP Tasks. Kazuma Hashimoto, Caiming Xiong, Yoshimasa Tsuruoka, and Richard Socher. The 2017 Conference on Empirical Methods in Natural Language Processing (EMNLP 2017).
https://arxiv.org/abs/1611.01587 (Equation (4))

### CODE REVIEW

File: torch/nn/utils/rnn.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def pack_padded_sequence(input, lengths, batch_first=False):
     return PackedSequence(torch.cat(steps), batch_sizes)
 
 
-def pad_packed_sequence(sequence, batch_first=False):
+def pad_packed_sequence(sequence, padding_value=0.0, batch_first=False):
```
Comment (soumith) [inline torch/nn/utils/rnn.py]: can you make it `batch_first=False, padding_value=0.0`, so that it's backward compatible.

After that, this PR is good to go. thank you.

**Review by soumith — CHANGES_REQUESTED** [review]

File: torch/nn/utils/rnn.py (hunk 1 above)
Comment (hassyGo) [reply torch/nn/utils/rnn.py]: Thanks.
I've just committed the modified version.

**Comment by soumith** [comment]
thank you!

**Comment by hassyGo** [comment]
Cool! Thank you so much. Really helpful.
